### PR TITLE
Refactor for datajs code-base

### LINKED ISSUE #88: Release the library properly and document installation
## Acceptance criteria

* [ ] I have a major release re recent changes
* [ ] I can install it via package registry using both `NPM` and `Yarn`
* [ ] I have docs about how to install it in NodeJS environment
* [ ] I have docs about how to install it using `NPM` and `Yarn` but for browser environment, eg, when developing a React app
* [ ] I have docs about how to install it in browsers

### PR DESCRIPTION
**Pull request by risenW**
This PR is for #88 

* [x] Refactor data.js main script into submodules
* [x] Refactor code-base to ES6 specification
* [x] Refactor test suite from ava to mocha
* [x] Update build process for browser and node environment
* [x] Update docstrings for main classes
* [x] Update installation guides in README for developers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -419,9 +419,9 @@ Requirements:
 
 ### Test
```
Comment (rufuspollock) [inline README.md:420]: Do we still have automated (CI) tests for both browser and node?

File: datajs/src/browser-utils/utils.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
-const stream = require('stream')
+import { Readable } from 'stream'
+// import { File } from '../file-base'
```
Comment (rufuspollock) [inline datajs/src/browser-utils/utils.js]: Why is this commented out? If we don't need it let's delete it.

File: datajs/src/data.js [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+export const KNOWN_TABULAR_FORMAT = ['csv', 'tsv', 'dsv']
+
+
+/**
+ * create a File from a pathOrDescriptor
+ * @param {string} pathOrDescriptor 
+ * @param {object} options 
+ */
+export function open(pathOrDescriptor, { basePath, format } = {}) {
+    let descriptor = null
```
Comment (rufuspollock) [inline datajs/src/data.js]: we should have 2-space indentation in JS. Can we set linter for this and fix that in these files. See https://playbook.datopian.com/style-guide/javascript/

File: datajs/test/browser-utils/utils.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
+// import path from 'path'
```
Comment (rufuspollock) [inline datajs/test/browser-utils/utils.js]: Why is this all commented out?

**Review by rufuspollock — CHANGES_REQUESTED** [review]
@risenW looks really good and would be good to approve except for the 4 space vs 2 spaces and a few misc questions?

File: README.md (hunk 1 above)
Comment (risenW) [reply README.md:420]: Yes, there is. `yarn test` runs all test for browser and node. `yarn test:node` and `yarn test:browser` for node and browser respectively

File: datajs/src/browser-utils/utils.js (hunk 2 above)
Comment (risenW) [reply datajs/src/browser-utils/utils.js]: Not needed. I'll delete it

File: datajs/src/data.js (hunk 3 above)
Comment (risenW) [reply datajs/src/data.js]: Okay, will do that now

File: datajs/test/browser-utils/utils.js (hunk 4 above)
Comment (risenW) [reply datajs/test/browser-utils/utils.js]: Not needed. These tests already exist in File Interface

File: datajs/src/data.js (hunk 3 above)
Comment (risenW) [reply datajs/src/data.js]: Fixed, pushed new changes already

**Comment by rufuspollock** [comment]
@risenW great work 👍. Merging.

**Pull request merged**